### PR TITLE
client: Add key settings for storing data

### LINKED ISSUE #56: Add a key setting to classify the stored properties.
## Feature Request

**Is your feature request related to a problem? Please describe.**  

Currently, only one sign can be stored in client.

**Describe the solution you'd like**  

When using the store function, we need to add a key setting to classify the stored properties.

**Describe alternatives you've considered**  

**Additional context**

### PR DESCRIPTION
**Pull request by HyeockJinKim**
# Summary 

- It closes #56

## How to Review? 

- Please refer to my self-review submitted as the review.

### CODE REVIEW

File: packages/client/src/index.ts [hunk 1]
```diff
@@ -11,25 +11,27 @@ export type DecryptedMessage = {
 };
 
 export function storeSignature(
+  key: string,
```
Comment (HyeockJinKim) [inline packages/client/src/index.ts:14]: The key value among the key and value of local storage.

File: packages/client/src/index.ts [hunk 2]
```diff
@@ -11,25 +11,27 @@ export type DecryptedMessage = {
...
   properties: Property[],
   sign: Signature,
   storage: StorageType,
   password?: string,
 ): void {
-  const key = getGeneratedRawKey(storage, password);
-  const encrypted = encryptMessage(JSON.stringify({ properties, sign }), key);
-  storeData(encrypted, DataType.Message, storage);
+  const encryptionKey = getGeneratedRawKey(key, storage, password);
```
Comment (HyeockJinKim) [inline packages/client/src/index.ts:20]: Change the name to distinguish the encryption key and the key among the key and value.

File: packages/client/src/utils.ts [hunk 3]
```diff
@@ -27,21 +27,26 @@ export function generateRandomKey(): string {
...
 
-export function dataTypeToString(dataType: DataType): string {
+export function dataTypeToString(key: string, dataType: DataType): string {
   switch (dataType) {
     case DataType.Key:
-      return 'Key';
+      return `Key_${key}`;
     case DataType.Message:
-      return 'Message';
+      return `Message_${key}`;
```
Comment (HyeockJinKim) [inline packages/client/src/utils.ts:35]: Store the value through the key separately.

File: packages/client/test/client.spec.ts [hunk 4]
```diff
@@ -42,12 +42,12 @@ describe('test utils', () => {
...
-    storeData('stored key', DataType.Key, StorageType.LocalStorage);
-    storeData('stored message', DataType.Message, StorageType.LocalStorage);
-    expect(getData(DataType.Key, StorageType.LocalStorage).get()).to.be.equal('stored key');
-    expect(getData(DataType.Message, StorageType.LocalStorage).get()).to.be.equal('stored message');
+    expect(getData('test', DataType.Key, StorageType.LocalStorage).orUndefined()).to.be.equal(undefined);
+    expect(getData('test', DataType.Message, StorageType.LocalStorage).orUndefined()).to.be.equal(undefined);
+    storeData('test', 'stored key', DataType.Key, StorageType.LocalStorage);
+    storeData('test', 'stored message', DataType.Message, StorageType.LocalStorage);
+    expect(getData('test', DataType.Key, StorageType.LocalStorage).get()).to.be.equal('stored key');
+    expect(getData('test', DataType.Message, StorageType.LocalStorage).get()).to.be.equal('stored message');
```
Comment (HyeockJinKim) [inline packages/client/test/client.spec.ts:50]: In the test case, `test` value is assigned as the key.

File: packages/client/test/client.spec.ts [hunk 5]
```diff
@@ -95,20 +95,35 @@ describe('test store localStorage', () => {
...
+    storeSignature('test2', properties, sign, StorageType.LocalStorage);
+    const storageData1 = getSignature('test1', StorageType.LocalStorage, 'abc');
+    expect(storageData1).to.be.not.null;
+    expect(storageData1?.properties).to.be.deep.equal(properties);
+    expect(storageData1?.sign).to.be.deep.equal(sign);
+    const storageData2 = getSignature('test2', StorageType.LocalStorage);
+    expect(storageData2).to.be.not.null;
+    expect(storageData2?.properties).to.be.deep.equal(properties);
+    expect(storageData2?.sign).to.be.deep.equal(sign);
+    expect(() => getSignature('test2', StorageType.LocalStorage, 'abc')).to.throw(Error);
```
Comment (HyeockJinKim) [inline packages/client/test/client.spec.ts:126]: Test case to check whether the value is stored separately if the key value is different.

**Review by HyeockJinKim — COMMENTED** [review]
self review

File: packages/client/src/index.ts [hunk 6]
```diff
@@ -11,25 +11,34 @@ export type DecryptedMessage = {
...
+  key: string,
   storage: StorageType,
   password?: string,
 ): DecryptedMessage | null {
-  const key = getGeneratedRawKey(storage, password);
-  const decryptedMessage = getDecryptedMessage(key, storage);
-  return decryptedMessage.matches({
-    present: (message) => JSON.parse(message) as DecryptedMessage,
-    empty: () => null
+  const encryptionKey = getData(key, DataType.Key, storage);
```
Comment (HyeockJinKim) [inline packages/client/src/index.ts:30]: When reading the stored signature, a key is not created.

**Review by anthonyminyungi — APPROVED** [review]

**Pull request merged**